auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LDAPCacheKey.java

```diff
@@ -0,0 +1,58 @@
+package com.redhat.lightblue.rest.auth.ldap;
+
+import javax.naming.directory.SearchControls;
+import javax.naming.ldap.LdapContext;
+
+/**
+ * This class holds the values to represent an unique cache key. It is also used to query data as well
+ *
+ * Created by lcestari on 2/23/15.
+*/
+public class LDAPCacheKey {
+    final String uid;
+    final LdapContext ldapContext;
+    final String ldapSearchBase;
+    final String searchFilter;
+    final SearchControls searchControls = new SearchControls();
+
+    public LDAPCacheKey(String uid, LdapContext ldapContext, String ldapSearchBase, String searchFilter) {
+        this.uid = uid;
+        this.ldapContext = ldapContext;
+        this.ldapSearchBase = ldapSearchBase;
+        this.searchFilter = searchFilter;
+        this.searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
```

Comment:
Provide a constructor that takes search scope as an argument so it could be changed?  In our deployment this makes sense, will it make sense everywhere?